thrimshim/thrimshim/main.py

```diff
@@ -73,11 +75,40 @@ def get_row(ident):
 	}
 	return json.dumps(response)
 
+def get_all_rows():
+	"""Gets all rows from the events table from the database"""
+	conn = app.db_manager.get_conn()
+	results = database.query(conn, """
+		SELECT *
+		FROM events""")
+	
+	rows = []
+	row = results.fetchone()
+	while row:
+		row = row._asdict()
+		row['id'] = str(row['id'])
+		row = {
+			key: (
+				value.isoformat() if isinstance(value, datetime.datetime)
+				else value
+			) for key, value in row.items()
+		}
+		rows.append(row)
+		row = results.fetchone()
+	if not rows:
+		return 'No rows found', 404
+	
+	response = rows
+	return json.dumps(response)
+
 
 def update_row(ident, new_row):
 	"""Updates row of database with id = ident with the edit columns in
 	new_row."""
 
+	if not authenticate_user(new_row['Auth_Token']):
```

Comment:
I don't know if this is easier or harder on the client side, but generally I'd expect this kind of auth token to go in an `Authorization` header, not in the post data.
There's an advantage there in that most standard tools know not to log that field, treat it confidentially, etc.

Reply:
That comes from me not really knowing what I was doing, and me copying what I did last time (which was both a hackjob, and worked completely differently with authing on page load and generating session keys). Maybe just take this out for now?

Reply:
Sure, we can address later.